main.rb

```diff
@@ -0,0 +1,98 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    size = self.size
+    while i < size
+      is_a?(Range) ? yield(min + i) : yield(self[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    idx = 0
+    my_each do |elem|
+      yield(elem, idx)
+      idx += 1
+    end
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    if is_a?(Array)
+      selected = []
+      my_each { |ele| selected << ele if yield ele }
+    else
+      selected = {}
+      my_each { |key, value| selected[key] = value if yield(key, value) }
+    end
+    selected
+  end
+
+  # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+
+  def my_all?(arg = nil, &prc)
+    return true if !block_given? && arg.nil? && include?(nil) == false && include?(false) == false
+    return false unless block_given? || !arg.nil?
+
+    if block_given?
+      my_each { |elem| return false if prc.call(elem) == false }
+    elsif arg.class == Regexp
+      my_each { |elem| return false if arg.match(elem).nil? }
+    elsif arg.class <= Numeric || arg.class <= String
+      my_each { |elem| return false if elem != arg }
+    else
+      my_each { |elem| return false if (elem.is_a? arg) == false }
+    end
+    true
+  end
+
+  def my_any?(arg = nil, &prc)
+    return true if !block_given? && arg.nil? && my_each { |elem| return true if elem == true } && empty? == false
+    return false unless block_given? || !arg.nil?
+
+    my_each do |ele|
+      return true if prc.call(ele) == true
+    end
+    false
+  end
+
+  def my_none?(arg = nil, &prc)
+    !my_any?(arg, &prc)
+  end
+
+  def my_count
+    count = 0
+    my_each do |_|
+      count += 1
+    end
+    count
+  end
```

Comment:
#### my_count

- [ ] counts the number of items in enum that are equal to the item if an argument is given

```ruby
[1, 1, 2 ,3].my_count(1)  => 2
```

- Your my_count doesn't take arguments and that's why is given an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/77498346-04ac7100-6e15-11ea-8c60-164ed7be7e5a.png)

- Real count
![image](https://user-images.githubusercontent.com/52209067/77498374-1beb5e80-6e15-11ea-809a-002f935ef4ae.png)

- [ ] counts the number of elements yielding a true value if a block is given

```ruby
[1, 2, 3].my_count {|i| i > 1} => 2
```

- Yours is giving the wrong answer
![image](https://user-images.githubusercontent.com/52209067/77498459-4b9a6680-6e15-11ea-9702-a27574a484c6.png)

- Real count
![image](https://user-images.githubusercontent.com/52209067/77498482-5a811900-6e15-11ea-953a-c1793557df73.png)